--- a/libseq/Makefile
+++ b/libseq/Makefile
@@ -19,7 +19,7 @@
 	$(MAKE) FPIC="$(FPIC_OPT)" libmpiseq$(PLAT)$(LIBEXT_SHARED)
 
 libmpiseq$(PLAT)$(LIBEXT_SHARED): mpi.o mpic.o elapse.o
-	$(FC) -shared mpi.o mpic.o elapse.o -o libmpiseq$(PLAT)$(LIBEXT_SHARED)
+	$(FC) -shared -Wl,$(SONAME)libmpiseq$(PLAT)$(LIBEXT_SHARED) mpi.o mpic.o elapse.o -o libmpiseq$(PLAT)$(LIBEXT_SHARED)
 
 .f.o:
 	$(FC) $(OPTF) $(FPIC)              -c $*.f $(OUTF)$*.o
--- a/PORD/lib/Makefile
+++ b/PORD/lib/Makefile
@@ -29,7 +29,7 @@
 	$(RANLIB) $@
 
 libpord$(PLAT)$(LIBEXT_SHARED):$(OBJS)
-	$(CC) -shared $(OBJS) -o libpord$(PLAT)$(LIBEXT_SHARED)
+	$(CC) -shared -Wl,$(SONAME)libpord$(PLAT)$(LIBEXT_SHARED) $(OBJS) -o libpord$(PLAT)$(LIBEXT_SHARED)
 
 clean:
 	rm -f *.o
